towhee/hub/file_manager.py

```diff
@@ -329,22 +325,24 @@ def get_pipeline(self, pipeline: str, branch: str = 'main', redownload: bool = F
 
             if author == 'local':
                 if found_existing is False:
-                    # TODO: filip-halt
-                    # Error logging.
                     engine_log.info('Local file not found, has it been imported?')
-                    file_path = None
-            else:
-                if redownload:
-                    # TODO: filip-halt
-                    # Error logging.
-                    rmtree(file_path.parent)
+                    raise FileNotFoundError('Local file not found, has it been imported?')
+                return file_path
 
-                if not file_path.is_file():
-                    download_repo(author, repo, branch, str(file_path.parent), install_reqs=install_reqs)
+            if not file_path.is_file():
+                download_repo(author, repo, branch, str(file_path.parent), install_reqs=install_reqs)
+            else:
+                repo_path = str(path).rsplit('/', 1)[0]
+                repo_name = str(path).rsplit('/', 1)[1].split('.')[0]
+                repo = git.Repo(repo_path)
+                if not 'Your branch is up to date' in repo.git.status():
+                    print(f'Your local pipeline {repo_name} is not up to date, updating to latest version...')
+                    fetch_head = repo.remote('origin').fetch(branch)
+                    repo.git.reset(fetch_head, '--hard')
 
         return file_path
 
-    def get_operator(self, operator: str, branch: str = 'main', redownload: bool = False, install_reqs=True):
+    def get_operator(self, operator: str, branch: str = 'main', install_reqs=True):
```

Comment:
The same with the previous comment.

Reply:
Will create another PR to change all the `branch` to `tag`